Arrangement_on_surface_2/include/CGAL/Arr_bounded_planar_topology_traits_2.h

```diff
@@ -190,189 +191,162 @@ class Arr_bounded_planar_topology_traits_2 :
   //@}
 
 private:
-
   /// \name Auxiliary type definitions.
   //@{
-  typedef Arrangement_on_surface_2<Geometry_traits_2, Self>    Arr;
-
-  // Type definition for the constuction sweep-line visitor.
-  typedef Arr_construction_subcurve<Geometry_traits_2>         CSubcurve;
-  typedef Arr_construction_event<Geometry_traits_2, CSubcurve, Arr>
-                                                               CEvent;
-  typedef Arr_bounded_planar_construction_helper<Geometry_traits_2,
-                                                 Arr,
-                                                 CEvent,
-                                                 CSubcurve>    CHelper;
-
-  // Type definition for the basic insertion sweep-line visitor.
-  typedef Arr_basic_insertion_traits_2<Geometry_traits_2, Arr> BInsTraits;
-  typedef Arr_construction_subcurve<BInsTraits>                BISubcurve;
-  typedef Arr_construction_event<BInsTraits, BISubcurve, Arr>
-                                                               BIEvent;
-  typedef Arr_bounded_planar_insertion_helper<BInsTraits, Arr, BIEvent,
-                                              BISubcurve>      BIHelper;
-
-  // Type definition for the insertion sweep-line visitor.
-  typedef Arr_insertion_traits_2<Geometry_traits_2, Arr>       InsTraits;
-  typedef Arr_construction_subcurve<InsTraits>                 ISubcurve;
-  typedef Arr_construction_event<InsTraits, ISubcurve, Arr>
-                                                               IEvent;
-  typedef Arr_bounded_planar_insertion_helper<InsTraits, Arr, IEvent,
-                                              ISubcurve>       IHelper;
-
-  // Type definition for the batched point-location sweep-line visitor.
-  typedef Arr_batched_point_location_traits_2<Arr>             BplTraits;
-  typedef Arr_bounded_planar_batched_pl_helper<BplTraits, Arr> BplHelper;
+  typedef Arrangement_on_surface_2<Gt2, Self>                   Arr;
+
+public:
+  // The following definitions of helper functions use a C++11 features called
+  // alias template. This feature can be mimiked by non C++11 support simply by
+  // defining a type that inherits from the template we need to alias.  However,
+  // the non-C++11 code requires the (re)definition of all constructors of the
+  // derived class. The non-C++11 code follows the commented out C++11 code.
+  // When we move to C++11 we can use the more elgant code.
+#if 0
```

Comment:
There are some macros in CGAL to enable the code on platforms that support template aliases.